hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -2136,6 +2136,23 @@ public String getScmRatisRoles() {
         HddsUtils.format(server.getRatisRoles()) : "STANDALONE";
   }
 
+  @Override
+  public List<List<String>> getRatisRoles() {
+    final SCMRatisServer server = getScmHAManager().getRatisServer();
+    List<String> ratisRoles = server.getRatisRoles();
+    List<List<String>> result = new ArrayList<>();
+    for (String role : ratisRoles) {
+      String[] roleArr = role.split(":");
+      List<String> scmInfo = new ArrayList<>();
+      scmInfo.add(roleArr[0]);
+      scmInfo.add(roleArr[3]);
+      scmInfo.add(roleArr[1]);
+      scmInfo.add(roleArr[2]);
+      result.add(scmInfo);
+    }
```

Comment:
Seems `SCMRatisServerImpl#getRatisRoles` might return empty string if `RaftPeer#getAddress` is null (maybe in the case of SCM STANDALONE setup) which might trigger index exception. Let's handle it just in case, both here and in `scm-overview.html`.

Reply:
Thank you for review the code! I will continue to improve it.

Reply:
Agree, this should be handled, like it is for OM. https://github.com/apache/ozone/blob/c77115abd15778e9d4f00ec32d8d2b7a4a2ee670/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java#L3076-L3099